### PR TITLE
Adding support for flight and slice parameters to authorization request.  

### PR DESCRIPTION
**Pull request by shoatman**

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java [hunk 1]
```diff
@@ -89,6 +91,9 @@
     @SerializedName("x-client-DM")
     private String mDiagnosticDM;
 
+    protected transient AzureActiveDirectorySlice mSlice;
+
+    protected transient HashMap<String, String> mFlightParameters;
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java]: Does the concrete class matter here? Or can this be declared as `Map<String, String>`?

File: common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java [hunk 2]
```diff
@@ -164,6 +178,16 @@ public B setLibraryName(String libraryName) {
             return self();
         }
 
+        public B setSlice(AzureActiveDirectorySlice slice){
+            mSlice = slice;
+            return self();
+        }
+
+        public B setFlightParameters(HashMap<String, String> flightParameters){
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java]: Same question re `Map` vs `HashMap`

File: common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryOAuth2Configuration.java [hunk 3]
```diff
@@ -53,5 +56,21 @@ public void setAuthorityUrl(URL authorityUrl) {
         this.mAuthorityUrl = authorityUrl;
     }
 
+    public HashMap<String, String> getFlightParameters() {
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryOAuth2Configuration.java]: Would `Map<String, String>` suffice here?

File: common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationRequest.java [hunk 4]
```diff
@@ -151,10 +138,15 @@ public Uri getAuthorizationRequestAsHttpRequest() throws UnsupportedEncodingExce
             uriBuilder.appendQueryParameter(entry.getKey(), entry.getValue());
         }
 
-        for (Map.Entry<String, String> entry : ObjectMapper.deserializeQueryStringToMap(mSliceParameters).entrySet()) {
+        for (Map.Entry<String, String> entry : mFlightParameters.entrySet()) {
             uriBuilder.appendQueryParameter(entry.getKey(), entry.getValue());
         }
 
+        if(mSlice != null){
+            uriBuilder.appendQueryParameter("slice", mSlice.getSlice());
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationRequest.java]: nit: Can the same `"slice"` String constant be used here and in `AzureActiveDirectorySlice.java`?

File: common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java (hunk 1 above)
Comment (shoatman) [reply common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java]: Fixed.

File: common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java (hunk 2 above)
Comment (shoatman) [reply common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java]: Fixed.

**Pull request merged**